style_guide.md

```diff
@@ -0,0 +1,286 @@
+          },
+        ],
+        type: 'Radio',
+        },
+      ],
+    };
+    ```
+- In the `questionTitle` method we would return either of the two strings, based on the `isProxy` variable:
+    ```
+    local questionTitle(isProxy) = (
+      if isProxy == true then {
+      text: 'What was <em>{person_name_possessive}</em> age on their last birthday?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]} 
+      else 'What was your age on your last birthday?';
+    );
+    ```
+- Example of `questionTitle` with multiple variables:
+    ```
+    local questionTitle(isProxy, isEmployed) = (
+      if isProxy == true && isEmployed == true then {
+      text: 'What is the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]}
+      else if isProxy == false && isEmployed == true then {
+      text: 'What is the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]}
+      else if isProxy == true && isEmployed == false then 'What was the main activity of your organisation, business or freelance work?'
+      else {
+      text: 'What was the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+      placeholders: [
+        placeholders.personNamePossessive,
+      ]};
+    );  
+    ```
+- We would then do a similar thing for guidance with a `questionGuidance` method:
+    ```
+    local questionGuidance(isProxy) = (
+      if isProxy == true then 'Why we ask this question if they are retired or long-term sick or disabled' else 'Why we ask this question if you are retired or long-term sick or disabled';
+    );
+    ```
+### Block json
+
+- Block json can be a function if an argument is being passed:
+
+    ```
+    function(region_code, census_month_year_date) {
+      type: 'Question',
+      id: 'arrive-in-country',
+      question_variants: [
+        {
+          question: question(nonProxyTitle),
+          when: [rules.isNotProxy],
+        },
+        {
+          question: question(proxyTitle),
+          when: [rules.isProxy],
+        },
+      ],
+    } 
+    ```
+
+### Methods
+
+- We should define a method each time we’re changing a property.
+- Methods can be invoked at top-level or declared inline, depending on the intended use:
+
+    ```
+    // TOP-LEVEL
+        
+    local guidance(region_code, isProxy) = (
+      if region_code == 'GB-WLS' then
+        if isProxy then walesGuidanceProxy else walesGuidanceNonProxy
+      else if isProxy then englandGuidanceProxy else englandGuidanceNonProxy
+    );
+    
+    // INLINE (using region_code as top level argument, not external variable)
+    
+    local understandWelshBlock(region_code) = if region_code == 'GB-WLS' then [understand_welsh] else [];
+    ```
```

Comment:
I'm not sure we need this "Methods" section now that the variants examples use methods.

Reply:
Removed this whole "Methods" fragment. thanks!